darwin/dataset/upload_manager.py

```diff
@@ -1,528 +1,610 @@
-    dataset_id: int,
-    description: str,
-    expected_occurrences: List[int],
-    metadata: Optional[Dict],
-    name: str,
-):
-    """Create a new class on the dataset, with the parameters provided
-
-    Parameters
-    ----------
-    client: Client
-        Client authenticated to the team where the put request will be made
-    team: str
-        Team against which the client will make the requests
-    annotation_type_ids: List
-        List of the annotation types. TAG is ['1'].
-    cropped_image: Dict
-        Description of the thumbnail for the class. Can be None.
-        Example: {"image_id": 706170, "scale": 0.1455, "x": "2348", "y": "70"}
-    dataset_id: int
-        Dataset ID where to create the class
-    description: str
-        Class description. Can be empty string or None
-    expected_occurrences: List
-        List of integers denoting the expected occurrences.
-        Example: [0, 1]
-    metadata: Dict
-        Metadata for the class. Example {"_color": "auto"}
-    name: str
-        Class name
-
-    Returns
-    -------
-    response: Dict
-        Client response after the request
-    """
-    # Sanity checks and inject default values
-    if client is None:
-        raise ValueError(f"Client is {client}. Invalid value.")
-    if annotation_type_ids is None:
-        raise ValueError(f"Annotation type IDs is {annotation_type_ids}. Invalid value.")
-    if dataset_id is None:
-        raise ValueError(f"Dataset ID is {dataset_id}. Invalid value.")
-    if not description:
-        description = ""
-    if expected_occurrences is None:
-        raise ValueError(f"Expected occurrences is {expected_occurrences}. Invalid value.")
-    if not metadata:
-        metadata = {"_color": "auto"}
-    if not name:
-        raise ValueError(f"Class name is {name}. Invalid value.")
-
-    payload = {
-        "annotation_type_ids": annotation_type_ids,
-        "cropped_image": cropped_image,
-        "dataset_id": dataset_id,
-        "description": description,
-        "metadata": metadata,
-        "name": name,
-    }
-
-    endpoint = "annotation_classes"
-    response = client.post(endpoint=endpoint, payload=payload, team=team, retry=True)
-    if "errors" in response:
-        raise ValueError(f"Response error: {response['errors']}")
-    return response
+class LocalFile:
+    def __init__(self, local_path, **kwargs):
+        self.local_path = Path(local_path)
+        self.data = kwargs
+        self._type_check(kwargs)
+
+    def _type_check(self, args):
+        self.data["filename"] = args.get("filename") or self.local_path.name
+        self.data["type"] =  args.get("type") or self._infer_type()  
+
+    def _infer_type(self):
+        extension = Path(self.data["filename"]).suffix
+        if is_image_extension_allowed(extension):
+            return "image"
+        if is_video_extension_allowed(extension):
+            return "video"
+        raise UnsupportedFileType(self.data["filename"])
```

Comment:
Do we really need type checking? The backend is inferring the type already. (Still haven't read the whole PR while writing this, the type might be useful for something else)